pkg/synthetictests/disruption.go

```diff
@@ -276,3 +277,64 @@ func isOneSecondEvent(eventInterval monitorapi.EventInterval) bool {
 		return true
 	}
 }
+
+func dnsOverlapDisruption(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
```

Comment:
ah, I just noticed the test is never run.  Call it `testDnsOverlapDisruption`, then add it to the bottom of `StableSystemEventInvariants` and `SystemUpgradeEventInvariants`.  You will see several other synthetic tests called in those two places.  All of them have the same function signature as this one.